bindings/stronghold-nodejs/src/error.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity::account::Result as AccountResult;
+use identity::core::Result as CoreResult;
+use napi::bindgen_prelude::Error;
+use napi::Result;
+use serde_json::Result as SerdeResult;
+
+/// Convenience trait to simplify `result.map_err(account_error)` to `result.napi_result()`
+pub trait NapiResult<T> {
+  fn napi_result(self) -> Result<T>;
+}
+
+impl<T> NapiResult<T> for AccountResult<T> {
+  fn napi_result(self) -> Result<T> {
+    self.map_err(|account_error| Error::from_reason(account_error.to_string()))
+  }
+}
```

Comment:
Could we implement `NapiResult` for any `std::error::Error` or does that change the error message?

Does `error.to_string()` include the error variant name in the message? It might not which would make the messages less useful.

Reply:
Here's an example of error message for the `keySign` method: 
```
Invalid Proof Value: remote sign
```

Reply:
It looks like it omits the variant, it just happens to be in the error message:

```rust
  #[error("Invalid Proof Value: {0}")]
  InvalidProofValue(&'static str),
```

This may or may not be a problem based on how we choose to write error messages. The alternative is to append the error variant (which we can get via [`strum` deriving `IntoStaticStr`](https://docs.rs/strum/latest/strum/derive.IntoStaticStr.html)), similar to how it's used as the error name in the wasm bindings: https://github.com/iotaledger/identity.rs/blob/dev/bindings/wasm/src/error.rs

We might also need recursive printing of the error `source`s, like what #599 adds: https://github.com/iotaledger/identity.rs/blob/895a6fb768440cc364b2f75f3e3961dd679a597d/bindings/wasm/src/error.rs#L123

I consider improving the error messages for the Stronghold bindings optional since we can do it in a non-breaking change, this just needs to be recorded somewhere as a TODO I think.